src/components/table/table-cell/table-cell.component.js

```diff
@@ -19,14 +20,41 @@ import OptionsHelper from '../../../utils/helpers/options-helper';
  * set styling options for the cell used for action such as delete.
  */
 class TableCell extends React.Component {
+  /**
+   * Returns if child element has type property.
+   */
+  childHasType(child) {
+    return child && child.type;
+  }
+
+  /**
+   * Returns if child element is Textarea.
+   */
+  childrenHasTextArea() {
+    const { children } = this.props;
+
+    if (!Array.isArray(children)) {
+      return this.childHasType(children) && children.type.name === TextArea.name;
+    }
+
+    let result = false;
+    children.forEach((child) => {
+      if (this.childHasType(child) && child.type.name === TextArea.name) {
+        result = true;
+      }
+    });
+    return result;
+  }
+
   /**
    * Returns props to be used on the TD element.
    */
   get tableCellProps() {
     const { ...props } = validProps(this);
-
     delete props.children;
```

Comment:
Could this be achieved with destructuring on the line above?
`const { children, ...props } = validProps(this)`